bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/ScalingFontRegistry.java

```diff
@@ -0,0 +1,181 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2024 Yatta Solutions and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Yatta Solutions - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.graphics;
+
+import java.util.*;
+
+import org.eclipse.swt.internal.*;
+import org.eclipse.swt.internal.win32.*;
+import org.eclipse.swt.widgets.*;
+
+/**
+ * This class is used in the win32 implementation only to support
+ * scaling of fonts in multiple DPI zoom level.
+ * <p>
+ * <b>IMPORTANT:</b> This class is <em>not</em> part of the public
+ * API for SWT. It is marked public only so that it can be shared
+ * within the packages provided by SWT. It is not available on all
+ * platforms, and should never be called from application code.
+ * </p>
+ * @see <a href="http://www.eclipse.org/swt/">Sample code and further information</a>
+ * @noreference This class is not intended to be referenced by clients
+ */
+public class ScalingFontRegistry implements SWTFontRegistry {
+	private class ScaledFontContainer {
+		private Font baseFont;
```

Comment:
I would find it helpful to have some short explanation of the characteristics of this base font. Does it have a specific scale value? It is simply the one the font was first initialized with?

Reply:
It is the first one to be initialized, so usually it will always be scaled to the primary monitor zoom, unless its zoom is changed and a completely new font is created and used.

Reply:
I actually meant to find it helpful to have this information _in the code_, to ease comprehension when making modifications to the code :-)